File: Examples.VB/Features/Workbook/SaveWorkbookToCsvFile.vb
Namespace Features.Workbook
    Public Class SaveWorkbookToCsvFile
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim data = {
                {"Name", "City", "Birthday", "Sex", "Weight", "Height"},
                {"Bob", "NewYork", #6/8/1968#, "male", 80, 180},
                {"Betty", "NewYork", #7/3/1972#, "female", 72, 168},
                {"Gary", "NewYork", #3/2/1964#, "male", 71, 179},
                {"Hunk", "Washington", #8/8/1972#, "male", 80, 171},
                {"Cherry", "Washington", #2/2/1986#, "female", 58, 161},
                {"Eva", "Washington", #2/5/1993#, "female", 71, 180}
            }

            'Set data.
            Dim sheet As IWorksheet = workbook.Worksheets(0)
            sheet.Range("A1:F7").Value = data
            sheet.Tables.Add(sheet.Range("A1:F7"), True)
        End Sub
        Public Overrides ReadOnly Property CanDownload As Boolean
            Get
                Return True
            End Get
        End Property
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property
        Public Overrides ReadOnly Property SaveCsv As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace

File: Examples/Features/Workbook/SaveWorkbookToCsvFile.cs
﻿using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Workbook
{
    public class SaveWorkbookToCsvFile : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            object[,] data = new object[,]{
               {"Name", "City", "Birthday", "Sex", "Weight", "Height"},
               {"Bob", "NewYork", new DateTime(1968, 6, 8), "male", 80, 180},
               {"Betty", "NewYork", new DateTime(1972, 7, 3), "female", 72, 168},
               {"Gary", "NewYork", new DateTime(1964, 3, 2), "male", 71, 179},
               {"Hunk", "Washington", new DateTime(1972, 8, 8), "male", 80, 171},
               {"Cherry", "Washington", new DateTime(1986, 2, 2), "female", 58, 161},
               { "Eva", "Washington", new DateTime(1993, 2, 5), "female", 71, 180}
           };

            //Set data.
            IWorksheet sheet = workbook.Worksheets[0];
            sheet.Range["A1:F7"].Value = data;
            sheet.Tables.Add(sheet.Range["A1:F7"], true);
        }

        public override bool CanDownload
        {
            get
            {
                return true;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }

        public override bool SaveCsv
        {
            get
            {
                return true;
            }
        }
    }
}
